### PR TITLE
Add missing methods to logger collection

### LINKED ISSUE #1547: Metric aggragation is broken for LoggerCollection
## 🐛 Bug

After changes in #1278 it is now not possible to log testing metrics after traning while using several loggers.
### To Reproduce
Say we want to run a [MINST example](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/basic_examples/cpu_template.py) and also want to add a change - log testing metrics after training. For that we define a Callback

```
class TestCallback(Callback):
    def on_train_end(self, trainer, pl_module):
        # note that it would crash if you don't pass the `pl_module`
        trainer.test(pl_module)
```
and pass it to trainer `callbacks` argument. 

We would also like to use several loggers to track all metrics, say `MLFlowLogger` and `TensorBoardLogger`. For this we create instances of these loggers and pass them into `Trainer` in a list.

### Expected behavior
Testing metrics should be logged - but they don't as there's no final aggregation when our logger is a `LoggerCollection`

### Additional context
In my opinion, the logic in `agg_and_log_metrics`  and `_finalize_agg_metrics` is hard to follow, so I'd be happy if user could choose plain old `log_metrics` which worked nicely.

### PR DESCRIPTION
**Pull request by ethanwharris**
## What does this PR do?

Fixes #1547 

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@ethanwharris mind rebase on fixed master... :rabbit:

File: pytorch_lightning/loggers/base.py [hunk 1]
```diff
@@ -272,10 +272,20 @@ def __init__(self, logger_iterable: Iterable[LightningLoggerBase]):
     def __getitem__(self, index: int) -> LightningLoggerBase:
         return [logger for logger in self._logger_iterable][index]
 
+    def update_agg_funcs(
+            self,
+            agg_key_funcs: Optional[Mapping[str, Callable[[Sequence[float]], float]]] = None,
+            agg_default_func: Callable[[Sequence[float]], float] = np.mean
+    ):
+        [logger.update_agg_funcs(agg_key_funcs, agg_default_func) for logger in self._logger_iterable]
```
Comment (awaelchli) [inline pytorch_lightning/loggers/base.py:280]: codefactor marks this as an issue because the list is not use anywhere. perhaps write it without list comprehension.

Comment (Borda) [reply pytorch_lightning/loggers/base.py:280]: or just `_ = [...]`

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,6 +40,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Fixed TPU multi-core and Float16 ([#2632](https://github.com/PyTorchLightning/pytorch-lightning/pull/2632))
 
+- Fixed test metrics not being logged with multiple loggers ([#2723](https://github.com/PyTorchLightning/pytorch-lightning/pull/2723))
```
Comment (Borda) [inline CHANGELOG.md:43]:
```suggestion
- Fixed test metrics not being logged with `LoggerCollection` ([#2723](https://github.com/PyTorchLightning/pytorch-lightning/pull/2723))
```
